db/db_impl.cc

```diff
@@ -1164,12 +1165,15 @@ InternalIterator* DBImpl::NewInternalIterator(
   // Collect iterator for mutable mem
   merge_iter_builder.AddIterator(
       super_version->mem->NewIterator(read_options, arena));
-  std::unique_ptr<InternalIterator> range_del_iter;
+  std::unique_ptr<FragmentedRangeTombstoneIterator> range_del_iter;
   Status s;
   if (!read_options.ignore_range_deletions) {
+    auto read_seq = read_options.snapshot != nullptr
```

Comment:
can we pass down the sequence number chosen for the point-key iterators? I think data can be inconsistent if they're different.

This seqnum: https://github.com/facebook/rocksdb/blob/7125e2461934e43df312afb66e074f429ecbad43/db/db_impl.cc#L1778-L1781

Reply:
Ah good point. The two iterators use the same seqnum now.